manage_breast_screening/auth/views.py

```diff
@@ -48,3 +51,55 @@ def dev_sign_in(request):
 def sign_out(request):
     auth_logout(request)
     return redirect(reverse("home"))
+
+
+@login_not_required
+def cis2_sign_in(request):
+    """Start the CIS2 OAuth2/OIDC authorization flow."""
+    client = get_cis2_client()
+    redirect_uri = request.build_absolute_uri(reverse("auth:cis2_callback"))
+    return client.authorize_redirect(request, redirect_uri)
+
+
+@login_not_required
+def cis2_callback(request):
+    """Handle CIS2 OAuth2/OIDC callback, create/login the Django user, then redirect home."""
+    client = get_cis2_client()
+    token = client.authorize_access_token(request)
+    # Authlib parses id_token automatically for OIDC and stores userinfo
+    userinfo = token.get("userinfo") or {}
+    sub = userinfo.get("sub") or token.get("sub")
+    if not sub:
+        return HttpResponseBadRequest("Missing subject in CIS2 response")
+
+    User = get_user_model()
+    defaults = {}
+    if userinfo.get("email"):
+        defaults["email"] = userinfo["email"]
+    if userinfo.get("given_name"):
+        defaults["first_name"] = userinfo["given_name"]
+    if userinfo.get("family_name"):
+        defaults["last_name"] = userinfo["family_name"]
+
```

Comment:
Have you looked into [customising the user model](https://docs.djangoproject.com/en/5.2/topics/auth/customizing/#substituting-a-custom-user-model) yet? Now might be a good time to do so as we will probably want additional fields before long.